stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/usecase/SendMessageImpl.kt

```diff
@@ -15,20 +17,48 @@ public interface SendMessage {
      * @param message the message to send
      * @see io.getstream.chat.android.livedata.utils.RetryPolicy
      */
-    public operator fun invoke(message: Message, attachmentTransformer: ((at: Attachment, file: File) -> Attachment)? = null): Call<Message>
+    public operator fun invoke(
+        message: Message,
+        attachmentTransformer: ((at: Attachment, file: File) -> Attachment)? = null
+    ): Call<Message>
 }
 
 internal class SendMessageImpl(private val domainImpl: ChatDomainImpl) : SendMessage {
-    override operator fun invoke(message: Message, attachmentTransformer: ((at: Attachment, file: File) -> Attachment)?): Call<Message> {
+    override operator fun invoke(
+        message: Message,
+        attachmentTransformer: ((at: Attachment, file: File) -> Attachment)?
+    ): Call<Message> {
         val cid = message.cid
         validateCid(cid)
 
-        val channelController = domainImpl.channel(cid)
         return CoroutineCall(domainImpl.scope) {
+            val channelController = domainImpl.channel(cid)
+
+            populateMentions(message, channelController.toChannel())
+
             if (message.replyMessageId != null) {
                 channelController.replyMessage(null)
             }
             channelController.sendMessage(message, attachmentTransformer)
         }
     }
+
+    private fun populateMentions(
+        message: Message,
+        channel: Channel,
+    ) {
+        if (!message.text.contains('@')) {
+            return
+        }
+
+        val text = message.text.toLowerCase()
+        val mentions = channel.members.mapNotNull { member ->
+            if (text.contains("@${member.user.name.toLowerCase()}")) {
+                member.user.id
+            } else {
+                null
+            }
+        }
+        message.mentionedUsersIds = mentions.toMutableList()
```

Comment:
Copies the logic from the [Swift implementation](https://github.com/GetStream/stream-chat-swift/blob/e5bd6e22329e2da144fcdea1e6e6439c767fa60a/Sources/Client/Client/Client%2BChannel.swift#L277-L291) for now